File: app/src/main/java/com/lifedawn/bestweather/weathers/view/NotScrolledView.java
package com.lifedawn.bestweather.weathers.view;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Color;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;

import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;

import com.lifedawn.bestweather.R;
import com.lifedawn.bestweather.databinding.ViewNotScrolledBinding;

import java.util.Arrays;
import java.util.Collections;

public class NotScrolledView extends FrameLayout implements ICleaner {
	private ViewNotScrolledBinding binding;

	public NotScrolledView(Context context) {
		super(context);
		init(null);
	}

	public NotScrolledView(Context context, AttributeSet attrs) {
		super(context, attrs);
		init(attrs);
	}

	public NotScrolledView(Context context, AttributeSet attrs, int defStyleAttr) {
		super(context, attrs, defStyleAttr);
		init(attrs);
	}

	public NotScrolledView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
		super(context, attrs, defStyleAttr, defStyleRes);
		init(attrs);
	}

	private void init(@Nullable AttributeSet attrs) {
		int iconVisibility = View.GONE;
		String text = "ss";
		int iconSrc = R.drawable.temp_icon;

		TypedArray typedArray = getContext().getTheme().obtainStyledAttributes(attrs, R.styleable.NotScrolledView, 0, 0);
		try {
			iconVisibility = typedArray.getInt(R.styleable.NotScrolledView_iconVisibility, iconVisibility);
			text = typedArray.getString(R.styleable.NotScrolledView_android_text);
			iconSrc = typedArray.getResourceId(R.styleable.NotScrolledView_iconSrc, iconSrc);
		} catch (Exception e) {

		} finally {
			typedArray.recycle();
		}

		binding = ViewNotScrolledBinding.inflate(LayoutInflater.from(getContext()), this, true);
		binding.img.setVisibility(iconVisibility);
		binding.img.setImageResource(iconSrc);
		binding.text.setText(text);
		binding.text.setTextColor(Color.GRAY);
	}

	public void setImg(int id) {
		binding.img.setImageResource(id);
		binding.img.setVisibility(VISIBLE);
	}

	public void setText(String text) {
		binding.text.setText(text);
	}


	public void reDraw(int newX) {
		binding.getRoot().setX(newX);
		invalidate();
	}

	@Override
	protected void onLayout(boolean changed, int l, int t, int r, int b) {
		super.onLayout(changed, l, t, r, b);
	}

	@Override
	public void clear() {
		binding.img.setImageDrawable(null);
		removeAllViewsInLayout();
		binding = null;
	}
}

File: app/src/main/java/com/lifedawn/bestweather/ui/weathers/customview/NotScrolledView.kt
package com.lifedawn.bestweather.ui.weathers.customview

import android.content.Context
import android.graphics.Color
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.FrameLayout
import com.lifedawn.bestweather.R
import com.lifedawn.bestweather.databinding.ViewNotScrolledBinding

class NotScrolledView : FrameLayout, ICleaner {
    private var binding: ViewNotScrolledBinding? = null

    constructor(context: Context?) : super(context!!) {
        init(null)
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(context!!, attrs) {
        init(attrs)
    }

    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context!!, attrs, defStyleAttr) {
        init(attrs)
    }

    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(
        context!!, attrs, defStyleAttr, defStyleRes
    ) {
        init(attrs)
    }

    private fun init(attrs: AttributeSet?) {
        var iconVisibility = GONE
        var text = "ss"
        var iconSrc = R.drawable.temp_icon
        val typedArray = context.theme.obtainStyledAttributes(attrs, R.styleable.NotScrolledView, 0, 0)
        try {
            iconVisibility = typedArray.getInt(R.styleable.NotScrolledView_iconVisibility, iconVisibility)
            text = typedArray.getString(R.styleable.NotScrolledView_android_text)
            iconSrc = typedArray.getResourceId(R.styleable.NotScrolledView_iconSrc, iconSrc)
        } catch (e: Exception) {
        } finally {
            typedArray.recycle()
        }
        binding = ViewNotScrolledBinding.inflate(LayoutInflater.from(context), this, true)
        binding!!.img.visibility = iconVisibility
        binding!!.img.setImageResource(iconSrc)
        binding!!.text.text = text
        binding!!.text.setTextColor(Color.GRAY)
    }

    fun setImg(id: Int) {
        binding!!.img.setImageResource(id)
        binding!!.img.visibility = VISIBLE
    }

    fun setText(text: String?) {
        binding!!.text.text = text
    }

    fun reDraw(newX: Int) {
        binding!!.root.x = newX.toFloat()
        invalidate()
    }

    override fun onLayout(changed: Boolean, l: Int, t: Int, r: Int, b: Int) {
        super.onLayout(changed, l, t, r, b)
    }

    override fun clear() {
        binding!!.img.setImageDrawable(null)
        removeAllViewsInLayout()
        binding = null
    }
}